cdk/bin/resize.sh

```diff
@@ -0,0 +1,26 @@
+
+# Specify the desired volume size in GiB as a command-line argument. If not specified, default to 20 GiB.
+SIZE=${1:-20}
+
+# Install the jq command-line JSON processor.
+sudo yum -y install jq
+
+# Get the ID of the envrionment host Amazon EC2 instance.
+INSTANCEID=$(curl http://169.254.169.254/latest/meta-data//instance-id)
+
+# Get the ID of the Amazon EBS volume associated with the instance.
+VOLUMEID=$(aws ec2 describe-instances --instance-id $INSTANCEID | jq -r .Reservations[0].Instances[0].BlockDeviceMappings[0].Ebs.VolumeId)
+
+# Resize the EBS volume.
+aws ec2 modify-volume --volume-id $VOLUMEID --size $SIZE
+
+# Wait for the resize to finish.
+while [ "$(aws ec2 describe-volumes-modifications --volume-id $VOLUMEID --filters Name=modification-state,Values="optimizing","completed" | jq '.VolumesModifications | length')" != "1" ]; do
+  sleep 1
+  done
+
+# Rewrite the partition table so that the partition takes up all the space that it can.
+sudo growpart /dev/nvme0n1p1 1
```

Comment:
This fails when I try to run it due to:

```
~/environment/workshop (master) $ sudo growpart /dev/nvme0n1p1 1
FAILED: unable to determine partition type

```

Reply:
It seems like just waiting for the EBS volume to finish resizing, running resize2fs, and rebooting worked.

Reply:
Could we just do an end-run around all of this and make the original Cloud 9 instance create with a bigger volume?

Reply:
Looked more at: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/recognize-expanded-volume-linux.html

I think that the partition was not being referenced correctly.